lib/cadet_web/controllers/incentives_controller.ex

```diff
@@ -2,20 +2,40 @@ defmodule CadetWeb.IncentivesController do
   use CadetWeb, :controller
 
   use PhoenixSwagger
+  import Cadet.Assessments
 
+  alias Cadet.Accounts.CourseRegistrations
   alias Cadet.Incentives.{Achievements, Goals}
 
   def index_achievements(conn, _) do
     course_id = conn.assigns.course_reg.course_id
     render(conn, "index_achievements.json", achievements: Achievements.get(course_id))
   end
 
+  @spec index_goals(Plug.Conn.t(), any) :: Plug.Conn.t()
   def index_goals(conn, _) do
     render(conn, "index_goals_with_progress.json",
       goals: Goals.get_with_progress(conn.assigns.course_reg)
     )
   end
 
+  def calculate_totalxp(conn, _) do
+    course_id = conn.assigns.course_reg.course_id
+    total_achievement_xp = Achievements.get_total_xp(course_id)
```

Comment:
I'm quite confused. Is this supposed to be the total possible achievement XP or the total received achievement XP of the user?

On line 29, this is added to the user's total received assessment XP, so I presume you meant it to be the latter. But you only pass `course_id` to `get_total_xp`. That can't get you the total XP received by the user...